include/pistache/pist_timelog.h

```diff
@@ -152,7 +153,13 @@ class __PS_TIMEDBG
             {
                 for(unsigned int i=0; i<10; i++)
                     marker_chars[i] = the_marker;
+                #ifdef _IS_BSD
+                strlcpy(&(marker_chars[10]), "...", 4);
+                #elif defined(__linux__)
+                strncpy(&(marker_chars[10]), "...", 4);
+                #else
                 strcpy(&(marker_chars[10]), "...");
+                #endif
```

Comment:
I'm not a fan of using ifdefs anywhere. Any reason why you couldn't use strncpy unconditionally? If yes, what about implementing a wrapper "c_strcpy" function which calls whatever libc function which doesn't cause warnings? Also, note that `strlcpy` and `strlcat` are now part of POSIX.1-2024, and have been hence added to glibc.

Reply:
Well, Ubuntu 22.04 (still the latest LTS release) uses gcc (Ubuntu 11.4.0-1ubuntu1~22.04), which supports strncpy, but not strlcpy.

Conversely, BSD supports strlcpy but not strncpy.

macOS supports both.

As to ifdefs, they can perform wonderful magic, but I agree with you that it is best not to see ifdefs scattered through the code. I will isolated these differences so they don't appear in the middle of the function.